File: frontend/mi-vite/src/services/ninoService.js
import api from "../axios/axios";

export const listarNinos = async (idCliente) => {

    const response = await api.get(`/ninos/cliente/${idCliente}`);

    return response.data;

};

export const registrarNino = async (nino) => {

    const response = await api.post("/ninos", nino);

    return response.data;

};

export const actualizarNino = async (id, nino) => {

    const response = await api.put(`/ninos/${id}`, nino);

    return response.data;

};

export const eliminarNino = async (id) => {

    await api.delete(`/ninos/${id}`);

};

File: backend/backend/yupi/src/main/java/service/NinoService.java
package service;

import model.dto.Cliente.ClienteRequest;
import model.dto.Cliente.ClienteResponse;
import model.dto.Nino.NinoRequest;
import model.dto.Nino.NinoResponse;

public interface NinoService {

    NinoResponse registrarNino(NinoRequest ninoRequest);
    NinoResponse consultarNino(Integer id, NinoResponse ninoResponse);
    NinoResponse actualizarNino(Integer id, NinoRequest ninoRequest);
    void eliminarNino(Integer id);
    NinoResponse consultarNinoporCliente(Integer idCliente);
    NinoResponse consultarNinoporCumpleano(Integer idCumnpleano);

}
